app/models/endpoint.rb

```diff
@@ -0,0 +1,8 @@
+##
+# Metadata about a replication endpoint, including a unique human
+# readable name, and the type of endpoint it is (e.g. :online, :archive).
+class Endpoint < ApplicationRecord
+  has_many :preservation_copies
+  validates :endpoint_name, presence: true, uniqueness: true
+  validates :endpoint_type, presence: true
```

Comment:
Do we expect 'endpoint_type' to be a finite set and if so does it make sense to represent it as such?

Reply:
i think we do, but per #46, there will be a table containing a list of endpoint types.  otherwise, i'd agree that we should constrain the value to known good types.